registry/handlers/blobupload.go

```diff
@@ -30,8 +30,12 @@ func blobUploadDispatcher(ctx *Context, r *http.Request) http.Handler {
 
 	if !ctx.readOnly {
 		handler["POST"] = http.HandlerFunc(buh.StartBlobUpload)
+		if r.FormValue("digest") != "" && r.ContentLength > 0 {
+			handler["POST"] = http.HandlerFunc(buh.BlobUploadComplete)
```

Comment:
An alternative would be to check this condition in `StartBlobUpload` and call `BlobUploadComplete` from there. That feels a little tidier, even though it has the same effect.

I'm fine with either solution, just wanted to point out the alternative since it feels a little weird to set up a different handler based on the details of the request.